bin/install_swiftpm_dependencies

```diff
@@ -1,9 +1,67 @@
 #!/bin/bash -eu
 
+CUSTOM_PACKAGE_RESOLVED_PATH=${1-}
+BUILD_TYPE=${2-}
+SPM_CACHE_LOCATION="$HOME/spm"
+
+WORKSPACE_PACKAGE_RESOLVED_PATH=(*.xcworkspace/xcshareddata/swiftpm/Package.resolved)
+ROOT_PACKAGE_RESOLVED_PATH=Package.resolved
+
+# Find where Package.resolved is located
+if [[ -n "$CUSTOM_PACKAGE_RESOLVED_PATH" ]]; then
+  PACKAGE_RESOLVED_LOCATION=$CUSTOM_PACKAGE_RESOLVED_PATH
+elif  [[ -f ${WORKSPACE_PACKAGE_RESOLVED_PATH[0]} ]]; then
+  BUILD_TYPE="XCODEBUILD"
+  PACKAGE_RESOLVED_LOCATION=${WORKSPACE_PACKAGE_RESOLVED_PATH[0]}
+elif  [[ -f $ROOT_PACKAGE_RESOLVED_PATH ]]; then
+  BUILD_TYPE="SWIFT"
+  PACKAGE_RESOLVED_LOCATION=$ROOT_PACKAGE_RESOLVED_PATH
+else
+  echo "Unable to find Package.resolved"
+fi
+
+PACKAGE_RESOLVED_HASH=$(hash_file "$PACKAGE_RESOLVED_LOCATION")
+CACHE_KEY="$BUILDKITE_PIPELINE_SLUG-spm-cache-$PACKAGE_RESOLVED_HASH"
+
+# Restore SPM cache if it's available
+mkdir -p "$SPM_CACHE_LOCATION"
+cd "$SPM_CACHE_LOCATION"
+restore_cache "$CACHE_KEY"
+cd -
+
+# This will let Xcode use the system SSH config for downloading packages
 sudo defaults write com.apple.dt.Xcode IDEPackageSupportUseBuiltinSCM YES
 
 # Trust all GitHub.com and BitBucket.org keys – this allows checking out dependencies via SSH
 add_host_to_ssh_known_hosts bitbucket.org
 add_host_to_ssh_known_hosts github.com
 
-xcodebuild -resolvePackageDependencies
+# Based on the project type, resolve the packages using the correct build type
+if [[ "$BUILD_TYPE" == "XCODEBUILD" ]]; then
+  echo "	Resolving packages with \`xcodebuild\`"
```

Comment:
What do you think about [using the `~~~` "section" syntax that Buildkite supports](https://buildkite.com/docs/pipelines/managing-log-output) to control log output here?
(And probably same on the lines below (L46, L60, L64) too?)

On the other hand, that will create a new section while maybe we want all of those sub-messages to be grouped under the same parent… 🤷

Reply:
I thought about that and decided to keep them grouped under the "Restoring SPM Packages" parent for now. I feel like they aren't important enough to surface higher, but we can always easily change that if we decide differently in the future 🙂